packages/roosterjs-editor-plugins/lib/Paste/officeOnlineConverter/convertPastedContentFromWordOnline.ts

```diff
@@ -0,0 +1,227 @@
+            currentListItems.forEach((item) => {
+                insertListItem(convertedListElement, item, listType);
+            })
+        });
+
+        insertConvertedListToDoc(convertedListElement, doc.body, itemBlock);
+
+        const parentContainer = itemBlock.startElement.parentElement;
+        if (parentContainer) {
+            itemBlock.listItemContainers.forEach((listItemContainer) => {
+                parentContainer.removeChild(listItemContainer);
+            });
+        }
+    });
+}
+
+/**
+ * Take all the list items in the document, and group the consecutive list times in a list block;
+ * @param doc pasted document that contains all the list element.
+ */
+function getListItemBlocks(doc: HTMLDocument): ListItemBlock[] {
+    const listElements = doc.getElementsByClassName(LIST_CONTAINER_ELEMENT_CLASS_NAME);
+    const result: ListItemBlock[] = [];
+    let curListItemBlock: ListItemBlock;
+    for (let i = 0; i < listElements.length; i++) {
+        let curItem = listElements[i];
+        if (!curListItemBlock) {
+            curListItemBlock = createListItemBlock(curItem)
+        } else {
+            const { listItemContainers } = curListItemBlock;
+            const lastItemInCurBlock = listItemContainers[listItemContainers.length - 1];
+            if (curItem.isEqualNode(lastItemInCurBlock.nextElementSibling)
+                || getFirstLeafNode(curItem) == getNextLeafSibling(doc.body, lastItemInCurBlock)) {
+                listItemContainers.push(curItem);
+                curListItemBlock.endElement = curItem
+            } else {
+                curListItemBlock.endElement = lastItemInCurBlock;
+                result.push(curListItemBlock);
+                curListItemBlock = createListItemBlock(curItem);
+            }
+        }
+    }
+
+    if (curListItemBlock.listItemContainers.length > 0) {
+        result.push(curListItemBlock);
+    }
+
+    return result;
+}
+
+/**
+ * Flatten the list items, so that all the consecutive list items are under the same parent.
+ * @param doc Root element of that contains the element.
+ * @param listItemBlock The list item block needed to be flattened.
+ */
+function flattenListBlock(rootElement: Element, listItemBlock: ListItemBlock) {
+    const collapsedListItemSections = collapseNodes(rootElement, listItemBlock.startElement, listItemBlock.endElement, true);
+    if (collapsedListItemSections.length > 1) {
+        collapsedListItemSections.forEach((section) => {
+            if (section.firstChild && section.firstChild.nodeName == 'DIV') {
+                unwrap(section)
+            }
+        })
+    }
+}
+
+/**
+ * Get the list type that the container contains. If there is no list in the container
+ * return null;
+ * @param listItemContainer Container that contains a list
+ */
+function getContainerListType(listItemContainer: Element): string {
+    if (listItemContainer.querySelector(UNORDERED_LIST_TAG_NAME)) {
```

Comment:
listItemContainer is selected by a more strict selector, so now we should already know the structure, and we can directly get list tag name from it ( or it's child?), don't use querySelector here since it has too much flexibility and get result from any level.

Reply:
Make sense